src/datasets/utils/file_utils.py

```diff
@@ -190,32 +169,52 @@ def cached_path(
 
     if is_remote_url(url_or_filename):
         # URL, so get it from the cache (downloading if necessary)
+        # HF HTTP URLs are prepared as to hf:// URLs
         url_or_filename, storage_options = _prepare_path_and_storage_options(
             url_or_filename, download_config=download_config
         )
-        # Pass HTTP storage_options to get_from_cache only if passed HTTP download_config.storage_options
-        if (
-            storage_options
-            and storage_options.keys() < {"http", "https"}
-            and not (download_config.storage_options and download_config.storage_options.keys() < {"http", "https"})
-        ):
-            storage_options = {}
-        output_path = get_from_cache(
-            url_or_filename,
-            cache_dir=cache_dir,
-            force_download=download_config.force_download,
-            proxies=download_config.proxies,
-            resume_download=download_config.resume_download,
-            user_agent=download_config.user_agent,
-            local_files_only=download_config.local_files_only,
-            use_etag=download_config.use_etag,
-            max_retries=download_config.max_retries,
-            token=download_config.token,
-            ignore_url_params=download_config.ignore_url_params,
-            storage_options=storage_options,
-            download_desc=download_config.download_desc,
-            disable_tqdm=download_config.disable_tqdm,
-        )
+        if url_or_filename.startswith("hf://"):
+            resolved_path = huggingface_hub.HfFileSystem(
+                endpoint=config.HF_ENDPOINT, token=download_config.token
```

Comment:
```suggestion
                token=download_config.token
```

This is already the default value in huggingface_hub (parsed from the same environment variable)